tempto-core/src/main/java/com/teradata/tempto/internal/query/QueryResultValueComparator.java

```diff
@@ -196,12 +203,21 @@ private static int floatingEqualWithTolerance(Object actual, Object expected, do
 
     private int singleFloatingEqual(Object actual, Object expected)
     {
-        return floatingEqualWithTolerance(actual, expected, MACHINE_EPSILON_FLOATING_32B);
+        return floatingEqualWithTolerance(actual, expected, getFloatTolerance(MACHINE_EPSILON_FLOATING_32B));
```

Comment:
make tolerance passed to `floatingEqualWithTolerance` be a maximum allowed fraction of expected value. Not absolute allowed variance.